types/d3-axis/d3-axis-tests.ts

```diff
@@ -149,14 +155,52 @@ const gTransition = gSelection.transition();
 gSelection.call(topAxis);
 gTransition.call(topAxis);
 
-const svgSelection: Selection<SVGSVGElement, any, any, any> = select<SVGSVGElement, any>('g');
+const svgSelection: Selection<SVGSVGElement, any, any, any> = select<SVGSVGElement, any>('svg');
 const svgTransition = svgSelection.transition();
 
 svgSelection.call(leftAxis);
 svgTransition.call(leftAxis);
 
+const pathSelection: Selection<SVGPathElement, any, any, any> = select<SVGPathElement, any>('path');
+const pathTransition = svgSelection.transition();
+
+// $ExpectError
+pathSelection.call(bottomAxis);
+// $ExpectError
+pathSelection.call(bottomAxis);
+
 const canvasSelection: Selection<HTMLCanvasElement, any, any, any> = select<HTMLCanvasElement, any>('canvas');
 const canvasTransition = canvasSelection.transition();
 
-// canvasSelection.call(rightAxis); // fails, incompatible context container element
-// canvasTransition.call(rightAxis); // fails, incompatible context container element
+// $ExpectError
+canvasSelection.call(rightAxis); // fails, incompatible context container element
+// $ExpectError
+canvasTransition.call(rightAxis); // fails, incompatible context container element
+
+// --------------------------------------------------------------------------
+// Strict Function Types
+// --------------------------------------------------------------------------
```

Comment:
Laudable idea to think of explicitly testing `strictFunctionTypes`. However, it will not work, as you found, since dts-lint is running from the minimally permitted TS version up to the latest version. So for anything below TS 2.6 the tests will not fail as expected, as `strictFunctionTypes` were not enforced.

So I would simply remove this section.